Tests/NIOSSLTests/NIOSSLIntegrationTest.swift

```diff
@@ -1038,6 +1038,35 @@ class NIOSSLIntegrationTest: XCTestCase {
             XCTFail("Encountered unexpected error: \(error)")
         }
     }
+    
+    func testUnprocessedDataOnReadPathBeforeClosing() throws {
+        let serverChannel = EmbeddedChannel()
+        let clientChannel = EmbeddedChannel()
+
+        let context = try configuredSSLContext()
+
+        try serverChannel.pipeline.addHandler(try NIOSSLServerHandler(context: context)).wait()
+        try clientChannel.pipeline.addHandler(try NIOSSLClientHandler(context: context, serverHostname: nil)).wait()
+
+        let addr = try SocketAddress(unixDomainSocketPath: "/tmp/whatever2")
+        let connectFuture = clientChannel.connect(to: addr)
+        serverChannel.pipeline.fireChannelActive()
+        try interactInMemory(clientChannel: clientChannel, serverChannel: serverChannel)
+        try connectFuture.wait()
+
+        // Ok, we're connected. Now we want to close the server, and have that trigger a client CLOSE_NOTIFY.
+        // After the CLOSE_NOTIFY create another chunk of data.
+        let _ = serverChannel.close()
+        
+        // Create a new chunk of data after the close
+        var clientBuffer = clientChannel.allocator.buffer(capacity: 5)
+        clientBuffer.writeStaticString("hello")
+        _ = try clientChannel.writeAndFlush(clientBuffer).wait()
+
+        // Manually take the client outbound buffer and hand it off to the server to validate that an error is not thrown.
+        let clientOutboundBuffer = try clientChannel.readOutbound(as: ByteBuffer.self)!
+        XCTAssertNoThrow(try serverChannel.writeInbound(clientOutboundBuffer))
```

Comment:
This could do with being a bit stricter about what we expect. If we replace this with `interactInMemory`, then what we expect to see is:

1. The server should read the data the client sent.
2. The server should be closed.
3. The client should be closed.
4. No errors in either pipeline.

Can we add validations for that?